tox.ini

```diff
@@ -71,6 +71,16 @@ deps =
   black==21.5b0
 commands = black {posargs} '../retworkx' '../tests'
 
+[testenv:stubs]
+basepython = python3
+envdir = .tox/stubs
+deps =
+  pytest
+  mypy==0.910
+  pytest-mypy-testing==0.0.8
+  dataclasses; python_version<'3.7'
+commands = python -m pytest '../stubs-tests'
```

Comment:
It seems a bit odd that the tests aren't in the directory `tests`, but need to have their own completely separate directory.  Perhaps it would be nicer to restructure such that you have a structure
```text
src/
  | ...
tests/
  | unit/
     | digraph/
     | generators/
     | ...
  | stub/
     | pydigraph_test.py
     | pygraph_test.py
```
?

Reply:
I agree it is weird, but I am trying to avoid tests using `unittest` with tests using `pytest`

Reply:
Having a structure like I suggested doesn't mean you have to mix tests for different runners - that's why I split the directories within the `tests` directory into two parts.  You can change the `tox`/`pytest` configurations to only search the correct subdirectories.

I don't feel strongly about this (I was only brought in to review the type hints) - it's a decision for you guys who actually maintain retworkx.